renku/service/controllers/utils/project_clone.py

```diff
@@ -16,46 +16,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Utilities for renku service controllers."""
-import shutil
-
 from renku.core.commands.clone import project_clone
-from renku.core.errors import RenkuException
 from renku.service.logger import service_log
-from renku.service.utils import make_project_path
 from renku.service.views.decorators import requires_cache
 
 
 @requires_cache
 def user_project_clone(cache, user_data, project_data):
     """Clones the project for a given user."""
-    local_path = make_project_path(user_data, project_data)
-    if local_path is None:
-        raise RenkuException("project not found")
+    if "project_id" in project_data:
+        project_data.pop("project_id")
 
     user = cache.ensure_user(user_data)
+    project = cache.make_project(user, project_data)
+    project.abs_path.mkdir(parents=True, exist_ok=True)
 
-    if local_path.exists():
-        shutil.rmtree(local_path)
-
-        for project in cache.get_projects(user):
-            if project.abs_path == local_path:
-                project.delete()
-
-        if "project_id" in project_data:
-            project_data.pop("project_id")
-
-    repo, initialized = project_clone(
+    repo, project.initialized = project_clone(
         project_data["url_with_auth"],
-        local_path,
+        project.abs_path,
         depth=project_data["depth"] if project_data["depth"] != 0 else None,
         raise_git_except=True,
         config={"user.name": project_data["fullname"], "user.email": project_data["email"],},
         checkout_rev=project_data["ref"],
     )
-    project_data["initialized"] = initialized
+    project.save()
 
     service_log.debug(f"project successfully cloned: {repo}")
-    service_log.debug(f"project folder exists: {local_path.exists()}")
+    service_log.debug(f"project folder exists: {project.exists()}")
```

Comment:
Shouldn't this be `project.abs_path.exists()` ?

Reply:
Project is a model instance with a method `.exists()` which ensures the correct filesystem path assembly and check based on the data found in the model instance.